exception boom of unit

let rec f (x : int) : empty {!boom} = !boom ()

let g (x : int) =
  using ({} @ unit) @ () run
    (match f x with {}) : int {!boom}
  with {
  | val x @ _ -> x
  | !boom _ @ _ -> !boom ()
  }
;;

container stdio ;;

using ({} @ unit) @ () run
  g 43
with {
 | val x @ _ -> print_value x
 | !boom _ @ _ -> print_string "Boom!\n"
}